### PR TITLE
Add method to close a modal

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

As reported by @jmaslate, there is currently no way to close a modal programmatically from the outside. 

## Approach and changes

- Add a `removeModal` method to the return value of `useModal`
- Add a `isModalOpen` boolean to the  return value of `useModal`
- Deprecate the `getModal` method as the modal configuration can be kept in state in userland code when needed.

**Note to reviewers:** I intend to rewrite the modal components for v3. I'll add comprehensive tests then, I promise 🙈

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Review by amelako — APPROVED** [review]

**Review by robinmetral — COMMENTED** [review]
Looks like a good improvement 👍 I checked some existing implementations of this use case (for example where we need to close the modal after a successful form submission) and it looks like we were simply conditionally rendering the modal, so not using its isOpen state at all 🙈

File: packages/circuit-ui/components/Modal/ModalContext.tsx [hunk 1]
```diff
@@ -28,11 +28,21 @@ import { Modal, ModalProps } from './Modal';
...
+   *
+   * If you need access to the `onClose` method or `isOpen` state of the modal,
+   * use the `removeModal` and `isOpen` context properties instead.
+   */
   getModal: () => ModalProps | null;
 };
 
 export const ModalContext = createContext<ModalContextValue>({
   setModal: () => {},
+  removeModal: () => {},
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply packages/circuit-ui/components/Modal/ModalContext.tsx:44]: From the PR description:

> **Note to reviewers**: I intend to rewrite the modal components for v3. I'll add comprehensive tests then.

🙈

File: packages/circuit-ui/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -25,10 +25,10 @@ import IS_IOS from '../../util/ios';
 import { isFunction } from '../../util/type-check';
 import useClickHandler from '../../hooks/use-click-handler';
 
-type OnClose = (event: MouseEvent | KeyboardEvent) => void;
+type OnClose = (event?: MouseEvent | KeyboardEvent) => void;
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply packages/circuit-ui/components/Modal/Modal.tsx:28]: When calling `onClose` programmatically, there's no `event` (at least not always).

File: packages/circuit-ui/components/Modal/ModalContext.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (robinmetral) [reply packages/circuit-ui/components/Modal/ModalContext.tsx:44]: 🙈 missed this, sorry

**Review by robinmetral — APPROVED** [review]

**Pull request merged**